### PR TITLE
Clear cookie with samesite=none

### PR DESCRIPTION
**Pull request by Gijsbertbas**
If frontend & backend are not samesite the Bearer cookie does not get cleared which causes endpoints allowing anonymous users to be unavailable in case an invalidated token is present.

This PR forces removing the cookie which I think is the desired effect in any 'samesite' situation.

See for more details this thread: https://github.com/symfony/symfony/pull/36173

### CODE REVIEW

File: src/Security/Guard/JWTTokenAuthenticator.php [hunk 1]
```diff
@@ -57,7 +57,7 @@ public function onAuthenticationFailure(Request $request, AuthenticationExceptio
         $response = parent::onAuthenticationFailure($request, $authException); // TODO: Change the autogenerated stub
 
         if ($authException instanceof InvalidTokenException || $authException instanceof ExpiredTokenException) {
-            $response->headers->clearCookie('BEARER');
+            $response->headers->clearCookie('BEARER', '/', null, false, true, 'none');
```
Comment (RonRademaker) [inline src/Security/Guard/JWTTokenAuthenticator.php:60]: Does this also remove the cookie if samesite is set to another value than `none`?

Comment (Gijsbertbas) [reply src/Security/Guard/JWTTokenAuthenticator.php:60]: removing a cookie is established by setting a cookie which is expired. the `clearCookie` method here does exactly that (see below)

so 'overwriting' the BEARER does not care about the existing cookie and its samesite settings 

image

Comment (RonRademaker) [reply src/Security/Guard/JWTTokenAuthenticator.php:60]: But I mean, if there is a cookie with samesite set to strict or lax, will setting an expired cookie with samesite set to none clear it?

Comment (Gijsbertbas) [reply src/Security/Guard/JWTTokenAuthenticator.php:60]: a cookie will replace a previous cookie if the name, path and domain are the same. samesite is not a matching field.

alternatively we can use the same $samesite config input here as used in the SuccessHandler when creating the cookie

**Review by RonRademaker — APPROVED** [review]
Please make sure scrutinizer builds pass before merging

**Pull request merged**